vep-mapping-pipeline/repeat_expansion_variants/pipeline.py

```diff
@@ -24,33 +25,56 @@ def none_to_nan(*args):
 
 def load_clinvar_data(clinvar_xml):
     """Load ClinVar data, preprocess, and return it as a Pandas dataframe."""
-    # Iterate through ClinVar XML records. Load all Microsatellite variants
-    variant_data = []  # Name, RCVaccession, GeneSymbol, HGNC_ID
+    # Iterate through ClinVar XML records
+    variant_data = []  # To populate the return dataframe (see columns below)
+    stats = defaultdict(int)
     for i, clinvar_record in enumerate(clinvar_xml_utils.ClinVarDataset(clinvar_xml)):
-        if i % 10000 == 0:
-            logger.info(f'Processed {i} records, have {len(variant_data)} variants')
-        if clinvar_record.measure and clinvar_record.measure.variant_type == 'Microsatellite':
-            # Extract gene symbol(s). Here and below, dashes are sometimes assigned to be compatible with the variant
-            # summary format which was used previously.
-            gene_symbols = clinvar_record.measure.preferred_gene_symbols
-            if not gene_symbols:
-                gene_symbols = ['-']
-            # Extract HGNC ID
-            hgnc_ids = clinvar_record.measure.hgnc_ids
-            hgnc_id = hgnc_ids[0] if len(hgnc_ids) == 1 and len(gene_symbols) == 1 else '-'
-            # Append data strings
-            for gene_symbol in gene_symbols:
-                variant_data.append([
-                    clinvar_record.measure.name,
-                    clinvar_record.accession,
-                    gene_symbol,
-                    hgnc_id
-                ])
+        if i and i % 100000 == 0:
+            total_repeat_expansion_variants = stats[clinvar_xml_utils.ClinVarRecordMeasure.MS_REPEAT_EXPANSION] + \
+                                              stats[clinvar_xml_utils.ClinVarRecordMeasure.MS_NO_COMPLETE_COORDS]
+            logger.info(f'Processed {i} records, collected {total_repeat_expansion_variants} repeat expansion variant '
+                        f'candidates')
+
+        # Skip a record if it does not contain variant information
+        if not clinvar_record.measure:
+            continue
+
+        # Repeat expansion events come in two forms: with explicit coordinates and allele sequences (CHROM/POS/REF/ALT),
+        # or without them. In the first case we can compute the explicit variant length as len(ALT) - len(REF). In the
+        # second case, which is more rare but still important, we have to resort to parsing HGVS-like variant names.
+        microsatellite_category = clinvar_record.measure.microsatellite_category
+        stats[microsatellite_category] += 1
```

Comment:
Good point! It indeed counts the `None`s, we're just not outputting them afterwards (no need). It's just easier to do the counting this way, rather than including a separate “is this a microsatellite” check